src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/InitialConsiderationHelper.java

```diff
@@ -0,0 +1,120 @@
+                                nullCheck(respondent.getValue().getResponseRespondentName()))).findFirst()
+                .orElse(RESPONDENT_MISSING);
+    }
+
+    /**
+     * Creates hearing detail section for Initial Consideration.
+     * Display details of the hearing with the earliest hearing date from the collection of hearings
+     * @param hearingCollection the collection of hearings
+     * @return return table with details of hearing
+     */
+    public static String getHearingDetails(List<HearingTypeItem> hearingCollection) {
+        if (hearingCollection == null){
+            return HEARING_MISSING;
+        }
+
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd MMM yyyy");
+
+        return hearingCollection.stream()
+                .filter(hearingTypeItem -> hearingTypeItem != null && hearingTypeItem.getValue() != null)
+                .map(HearingTypeItem::getValue)
+                .filter(hearing -> hearing.getHearingDateCollection() != null && !hearing.getHearingDateCollection().isEmpty()).min(Comparator.comparing(
+                        (HearingType hearing) ->
+                                getEarliestHearingDate(hearing.getHearingDateCollection()).orElse(LocalDate.now().plusYears(100))))
+                .map(hearing -> String.format(HEARING_DETAILS,
+                        getEarliestHearingDate(hearing.getHearingDateCollection()).map(formatter::format).orElse(""),
+                        hearing.getHearingType(),
+                        getHearingDuration(hearing)))
+                .orElse(HEARING_MISSING);
+    }
+
+    public static Optional<LocalDate> getEarliestHearingDate(List<DateListedTypeItem> hearingDates) {
+        return hearingDates.stream()
+                .filter(dateListedTypeItem -> dateListedTypeItem != null && dateListedTypeItem.getValue() != null)
+                .map(DateListedTypeItem::getValue)
+                .filter(hearingDate ->  hearingDate.getListedDate() != null && !hearingDate.getListedDate().isEmpty())
+                .map(hearingDateItem -> LocalDateTime.parse(hearingDateItem.getListedDate()).toLocalDate()).min(Comparator.naturalOrder());
+    }
+
+    /**
+     * Creates the jurisdiction section for Initial Consideration.
+     * @param jurisdictionCodes the list of jurisdiction codes assigned to the case
+     * @return jurisdiction code section
+     */
+    public static String generateJurisdictionCodesHtml(List<JurCodesTypeItem> jurisdictionCodes) {
+        if (jurisdictionCodes == null){
+            return "";
+        }
+        StringBuilder sb = new StringBuilder()
+                .append("<h2>Jurisdiction Codes</h2>")
+                .append("<a target=\"_blank\" href=\"https://intranet.justice.gov.uk/documents/2017/11/jurisdiction-list.pdf\">")
+                .append("View all jurisdiction codes and descriptors (opens in new tab)</a><br><br>");

```

Comment:
Can't we have this as a constant that we bring in given it's static data?

Reply:
done